### PR TITLE
Add agreed testing approach and how a good data migration should look to README

### PR DESCRIPTION
**Pull request by elcorbs**
As per the notes on https://ideaflip.com/b/abb8fb6xk3zs/

I would appreciate reviewers opinions on the following questions in particular
- Are any of the additions extraneous/ unnecessary?
- Do you feel that anything important and relevant for the README has been missed?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -79,10 +79,31 @@ Our staging and production environments are hosted by AWS. We would deploy to pr
...
-To run tests:
+### Run the tests
 ```sh
 $ make test
 ```
+### Agreed Testing Approach
+- Use nUnit and Moq
+- Always follow a TDD approach
+- Tests should be independent of each other
+- Gateway tests should test against a real database
```
Comment (Renny4Real) [inline README.md:90]: Should we mention here that the real database would be a real instance of the database set up in docker for testing (not the actual instance in AWS)?

File: README.md [hunk 2]
```diff
@@ -79,10 +79,31 @@ Our staging and production environments are hosted by AWS. We would deploy to pr
...
+
+## Data Migrations
+### A good data migration
+- 99.99999% available
+- Record failure logs
+- Automated
+- Reliable
+- As close to real time as possible
+- Observable monitoring in place
+- Should not affect any exiting databases
```
Comment (Renny4Real) [inline README.md:106]: Typo on existing

File: README.md [hunk 3]
```diff
@@ -79,10 +79,31 @@ Our staging and production environments are hosted by AWS. We would deploy to pr
...
+- Test coverage should never go down
+- All use cases should be covered by E2E tests
+- Optimise when test run speed starts to hinder development
+- Unit tests and E2E tests should run in CI
+- Test database schemas should match up with production database schema
+- Have integration tests which test from the PostgreSQL database to API Gateway
+
+## Data Migrations
+### A good data migration
+- 99.99999% available
```
Comment (bjpirt) [inline README.md:100]: I'm not sure this is specific to data migrations - more about the general requirement of system uptime

File: README.md (hunk 2 above)
Comment (bjpirt) [inline README.md:106]: Typo exiting -> existing

File: README.md (hunk 1 above)
Comment (Renny4Real) [reply README.md:90]: I feel saying "against a real database" might imply running the gateway tests against the actual Postgres DB in AWS

File: README.md (hunk 3 above)
Comment (Renny4Real) [inline README.md:100]: I thought we said 5 nines i.e 99.999% availability. Has this changed?

Comment (elcorbs) [reply README.md:100]: Makes sense, I'll remove it from here

File: README.md (hunk 1 above)
Comment (elcorbs) [reply README.md:90]: Yeah good point

File: README.md (hunk 3 above)
Comment (elcorbs) [reply README.md]: I've deleted this line now due to Bens comment

**Pull request merged**